### PR TITLE
Add Spectel gTest

### PR DESCRIPTION
**Pull request by scsides**

### CODE REVIEW

**Review by jessemapel — APPROVED** [review]
This is ready to be merged once the angle stuff gets merged

File: isis/tests/AngleTest.cpp [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  EXPECT_NEAR(angle.radians(), Isis::PI, TEST_EPSILON); 
+}
+
+TEST(AngleOperators, QDebug) {
+  // Stream qDebug output to a string and then do a string comparison?
+  Isis::Angle angle(45.0, Isis::Angle::Degrees);
+  QString qdebug_output;
+  QDebug debug(&qdebug_output);
+  debug << angle;
+  EXPECT_STREQ(qdebug_output.toLatin1(), "0.785398 <radians> (45 <degrees>) ");
```
Comment (SgStapleton) [inline isis/tests/AngleTest.cpp]: Did we decide whether it was better to do this or to check for the existence of substrings? Example: (EXPECT_TRUE(error.toString().toLatin1().contains("...") within a try catch block

File: isis/tests/AngleTest.cpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+TEST(AngleExceptions, LessThanNullAngle){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+  
+  try {
+    angle1 < Isis::Angle();
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the < operator." );
```
Comment (SgStapleton) [inline isis/tests/AngleTest.cpp]: Again, were we wanting to do a substring check...?

File: isis/tests/AngleTest.cpp [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    
+TEST(AngleExceptions, NullAngleLessThan){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+
+  try {
+    Isis::Angle() < angle1;
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the < operator." );
```
Comment (SgStapleton) [inline isis/tests/AngleTest.cpp]: (saa)

File: isis/tests/AngleTest.cpp [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+TEST(AngleExceptions, NullAngleLessThanOrEqual){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+
+  try {
+    Isis::Angle() <= angle1;
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the < operator." );
```
Comment (SgStapleton) [inline isis/tests/AngleTest.cpp]: (saa)

File: isis/tests/AngleTest.cpp [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+TEST(AngleExceptions, GreaterThanNullAngle){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+  
+  try {
+    angle1 > Isis::Angle();
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the > operator." );
```
Comment (SgStapleton) [inline isis/tests/AngleTest.cpp]: (saa)

File: isis/tests/AngleTest.cpp [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+TEST(AngleExceptions, GreaterThanOrEqualToNullAngle){
+  Isis::Angle angle1(30., Isis::Angle::Degrees );
+  
+  try {
+    angle1 >= Isis::Angle();
+    FAIL() << "Expected an error";
+   }
+   catch(Isis::IException &e) {
+     EXPECT_EQ(e.toString().toLatin1(), "**PROGRAMMER ERROR** Cannot compare a invalid angles with the > operator." );
```
Comment (SgStapleton) [inline isis/tests/AngleTest.cpp]: (saa)

**Comment by SgStapleton** [comment]
I'm a little confused. Which version of the AngleTest is going in? (vs. https://github.com/USGS-Astrogeology/ISIS3/pull/529)

**Comment by krlberry** [comment]
@SgStapleton #529's angle test went in earlier. I just resolved the merge conflicts with this branch.

**Pull request merged**